tests/geodesic_tests.rs

```diff
@@ -0,0 +1,419 @@
+            // todo: reduce allowed difference to 0.0
+            // todo: disallow sign difference
+            assert_delta!(items[10], result.2, 1e1, true, "lon2-out (result.2)", line_num);
+        }
+        if outmask & caps::AZIMUTH != 0 {
+            // record_delta(items[11], result.3, line_num, &mut entries[3]);
+            // todo: reduce allowed difference to 0.0
+            assert_delta!(items[11], result.3, 1e1, false, "azi2-out (result.3)", line_num);
+        }
+        if outmask & caps::DISTANCE != 0 {
+            // record_delta(items[12], result.4, line_num, &mut entries[4]);
+            // todo: reduce allowed difference to 0.0
+            assert_delta!(items[12], result.4, 1e-2, false, "s12-out (result.4)", line_num);
+        }
+        if outmask & caps::REDUCEDLENGTH != 0 {
+            // todo: reduce allowed difference to 0.0
+            // todo: disallow sign difference
+            // todo: remove exceptions
+            if !f64::is_nan(result.5) {
+                // record_delta(items[13], result.5, line_num, &mut entries[5]);
+                assert_delta!(items[13], result.5, 1e1, true, "m12-out (result.5)", line_num);
+            }
+        }
+        if outmask & caps::GEODESICSCALE != 0 {
+            // todo: reduce allowed difference to 0.0
+            // todo: disallow sign difference
+            // todo: remove exceptions
+            if !f64::is_nan(result.6) {
+                // record_delta(items[14], result.6, line_num, &mut entries[6]);
+                assert_delta!(items[14], result.6, 1e1, true, "M12-out (result.6)", line_num);
+            }
+            // todo: reduce allowed difference to 0.0
+            // todo: disallow sign difference
+            // todo: remove exceptions
+            if !f64::is_nan(result.7) {
+                // record_delta(items[15], result.7, line_num, &mut entries[7]);
+                assert_delta!(items[15], result.7, 1e1, true, "M21-out (result.7)", line_num);
+            }
+        }
+        if outmask & caps::AREA != 0 {
+            // todo: reduce allowed difference to 0.0
+            // todo: disallow sign difference
+            // todo: remove exceptions
+            if !f64::is_infinite(result.8) {
+                // record_delta(items[16], result.8, line_num, &mut entries[8]);
+                assert_delta!(items[16], result.8, 1e1, true, "S12-out (result.8)", line_num);
+            }
+        }
+    });
+
+    // delta_entries.lock().unwrap().iter().enumerate().for_each(|(i, entry)| {
+    //     println!("result.{} abs {:e} at line {}, rel {:e} at line {}, sign-change {}", i, entry.diff_abs, entry.line_abs, entry.diff_rel, entry.line_rel, entry.sign_change);
+    // });
+}
+
+// placeholder: Geodesic_GenDirectLine
+
+#[test]
+fn test_gen_inverse_azi_vs_cpp() {
+    // Format: this-in[_a _f] lat1 lon1 lat2 lon2 outmask result=a12 s12-out azi1-out azi2-out m12-out M12-out M21-out S12-out
```

Comment:
Only indirectly. Everything after the colon on that line is copy-pasted from the header line of the data file that test is reading from. I thought it might be helpful context, but I could certainly drop it, and/or add intermediate variables that match the Rust parameter names.